src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -187,7 +190,44 @@ public Transaction GetTransactionById(uint256 trxid)
         /// <inheritdoc/>
         public Transaction[] GetTransactionsByIds(uint256[] trxids)
         {
-            throw new NotImplementedException();
+            if (!this.TxIndex)
+            {
+                this.logger.LogTrace("(-)[NOT_INDEXED]:null");
+                return null;
+            }
+
+            Transaction[] txes = new Transaction[trxids.Length];
+
+            using (DBreeze.Transactions.Transaction transaction = this.DBreeze.GetTransaction())
+            {
+                transaction.ValuesLazyLoadingIsOn = false;
+
+                for (int i = 0; i < trxids.Length; i++)
+                {
+                    Row<byte[], byte[]> transactionRow = transaction.Select<byte[], byte[]>(TransactionTableName, trxids[i].ToBytes());
+                    if (!transactionRow.Exists)
+                    {
+                        this.logger.LogTrace("(-)[NO_TX_ROW]:null");
+                        return null;
+                    }
+
+                    Row<byte[], byte[]> blockRow = transaction.Select<byte[], byte[]>(BlockTableName, transactionRow.Value);
+
+                    if (!blockRow.Exists)
+                    {
+                        this.logger.LogTrace("(-)[NO_BLOCK]:null");
+                        return null;
```

Comment:
Shouldn't this be a `continue`?  If the transaction doesn't exist, then just log and continue onto the next one?

Reply:
No, it's a sign of a mistake. If requested tx is not found smth is wrong. 
Components should never ask for a transaction that is not a part of consensus chain